src/network.rs

```diff
@@ -140,10 +145,11 @@ impl Network {
                     let source = if let Some(section) = self.sections.remove(&source) {
                         section
                     } else {
-                        // Pre-split section does not exists due to being already merged
-                        // or split.
-                        debug!("Pre-split section {} not found", log::prefix(&source));
-                        continue;
+                        // Split can be triggered only by join or relocation,
+                        // which can happen at most once per section tick, so
+                        // a split can also happen at most once, so it should not
+                        // be possible to reach this line.
+                        panic!("Pre-split section {} not found", log::prefix(&source));
```

Comment:
this line could be reached when: the pre-split section got merged due to its sibling asked it to do so after emits the Action::split ?

so `being already merged` could happen, but `being already split` shall not?

Reply:
So something like this: section `P0` decides to merge with `P1`. Then `P1` receives a node which triggers a split. Then the actions are handled, so first `Action::Merge` is handled to merge `P0` and `P1` to `P`, Then `Action::Split` is handled, but `P1` is no longer there, thus panic. Is that right?

Reply:
If that is true, then both situation are valid and we should not panic in any of them.

Reply:
yes. so we shall not panic, but just in the comment we only mention about `already merged`  but not `already split` (which you already put  more comments to explain why.)